engine/src/main/java/io/zeebe/engine/processing/bpmn/container/EventSubProcessProcessor.java

```diff
@@ -43,91 +42,50 @@ public EventSubProcessProcessor(final BpmnBehaviors bpmnBehaviors) {
+      final ExecutableFlowElementContainer element, final BpmnElementContext completing) {
 
     variableMappingBehavior
-        .applyOutputMappings(context, element)
+        .applyOutputMappings(completing, element)
         .ifRightOrLeft(
-            ok -> {
-              eventSubscriptionBehavior.unsubscribeFromEvents(context);
-              stateTransitionBehavior.transitionToCompletedWithParentNotification(element, context);
-            },
-            failure -> incidentBehavior.createIncident(failure, context));
-  }
-
-  @Override
-  public void onCompleted(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
-    if (element.getOutgoing().isEmpty()) {
-      /* can be dropped during migration; after migration this is done as part of
-      stateTransitionBehavior.transitionToCompletedWithParentNotification(...)*/
-      stateTransitionBehavior.afterExecutionPathCompleted(element, context);
-    }
-    // call after here (unmigrated processor)
-    stateTransitionBehavior.takeOutgoingSequenceFlows(element, context);
-
-    stateBehavior.removeElementInstance(context);
+            ok ->
+                stateTransitionBehavior.transitionToCompletedWithParentNotification(
+                    element, completing),
+            failure -> incidentBehavior.createIncident(failure, completing));
   }
 
   @Override
-  public void onTerminating(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
+  public void onTerminate(
+      final ExecutableFlowElementContainer element, final BpmnElementContext terminating) {
 
-    eventSubscriptionBehavior.unsubscribeFromEvents(context);
+    incidentBehavior.resolveIncidents(terminating);
 
-    final var noActiveChildInstances = stateTransitionBehavior.terminateChildInstances(context);
+    final var noActiveChildInstances = stateTransitionBehavior.terminateChildInstances(terminating);
     if (noActiveChildInstances) {
-      stateTransitionBehavior.transitionToTerminated(context);
+      onChildTerminated(element, terminating, null);
     }
   }
 
   @Override
-  public void onTerminated(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
-
-    eventSubscriptionBehavior.publishTriggeredBoundaryEvent(context);
-
-    incidentBehavior.resolveIncidents(context);
-
-    stateTransitionBehavior.onElementTerminated(element, context);
-
-    stateBehavior.removeElementInstance(context);
-  }
-
-  @Override
-  public void onChildActivating(
+  public Either<Failure, ?> onChildActivating(
       final ExecutableFlowElementContainer element,
       final BpmnElementContext flowScopeContext,
-      final BpmnElementContext childContext) {}
+      final BpmnElementContext childContext) {
+    if (childContext.getBpmnElementType() != BpmnElementType.START_EVENT) {
+      return Either.right(null);
+    }
+
+    return variableMappingBehavior.applyInputMappings(flowScopeContext, element);
```

Comment:
@korthout do you think we need a comment here, which says why we actually do these input mapping here and have no activate command?